pkg/ui/src/views/cluster/containers/clusterOverview/index.tsx

```diff
@@ -25,6 +25,35 @@ import spinner from "assets/spinner.gif";
 import { refreshNodes, refreshLiveness } from "src/redux/apiReducers";
 import "./cluster.styl";
 
+interface ClusterNameProps {
+  clusterName: string;
+}
+
+function renderTitleAndHeader(props: ClusterNameProps) {
+  const { clusterName } = props;
+  const title = clusterName ?  "Cluster Overview | " + clusterName : "Cluster Overview";
+  return <React.Fragment>
+    <Helmet><title>{ title }</title></Helmet>
+    <h1>{ title }</h1>
+  </React.Fragment>;
+}
+
+const mapStateToClusterNameProps = createSelector(
```

Comment:
I find this [`mapStateToProps`](https://react-redux.js.org/using-react-redux/connect-mapstate) composition a little confusing but understand you are following the convention of the file. I prefer to see `createSelector` used to create reusable utilities for getting parts of the state. [`createStructuredSelector`](https://github.com/reduxjs/reselect#createstructuredselectorinputselectors-selectorcreator--createselector) I find is a more elegant tool to construct an object for _mapping state to props_. So this could be rewritten like this,

```js
// up by imports
import identity from "lodash.identity";
// also you'll need to import createStructuredSelector from reselect
import { createSelector, createStructuredSelector } from "reselect";

// create selectors
const getCachedData = createSelector(
  identity,
  (state: AdminUIState) => state.cachedData,
);
const getClusterData = createSelector(
  getCachedData,
  cachedData => cachedData.cluster.data,
);
const getClusterName = createSelector(
  getClusterData,
  clusterData => clusterData.cluster_name,
);

// mapStateToProps
const mapStateToClusterNameProps = createStructuredSelector({
  clusterName: getClusterName,
});
```

`identity` is a lodash function that return the value passed to it, so I use it when using `creatingSelector` that selects a root property of the state.